### ISSUE TITLE
Unsuitable value type Error 

### ISSUE THREAD

**Issue by anitagutta** [terraform-google-modules/terraform-google-bootstrap#165]
### TL;DR

problem brief:
- when executing terraform init on a TFE platform for creating a boot strap module which references to "terraform-google-modules/project-factory/google" it downloads .terraform/modules/"name"/modules/ and here there are many modules referencing to specific service and it will also have [version.tf](http://version.tf/) with version id as below against each module. 

terraform {
  required_version = ">= 0.13"
  required_providers {
    google = {
      source  = "hashicorp/google"
      version = ">= 3.50, < 4.0"
    }
    google-beta = {
      source  = "hashicorp/google-beta"
      version = ">= 3.50, < 4.0"
    }
    null = {
      source  = "hashicorp/null"
      version = "~> 2.1"
    }
    random = {
      source  = "hashicorp/random"
      version = "~> 2.2"

Below is the error we are seeing then terraform plan is run 

Same error is seen with TF version 0.12.16 and 0.13.15

![image](https://user-images.githubusercontent.com/97640419/176957704-23415c5c-4aa3-4ffc-bd14-eacd48e45e64.png)

### Expected behavior

Terraform plan create resources

### Observed behavior

![image](https://user-images.githubusercontent.com/97640419/176957704-23415c5c-4aa3-4ffc-bd14-eacd48e45e64.png)

### Terraform Configuration
terraform {
  required_version = ">= 0.13"
  required_providers {
    google = {
      source  = "hashicorp/google"
      version = ">= 3.50, < 4.0"
    }
    google-beta = {
      source  = "hashicorp/google-beta"
      version = ">= 3.50, < 4.0"
    }
    null = {
      source  = "hashicorp/null"
      version = "~> 2.1"
    }
    random = {
      source  = "hashicorp/random"
      version = "~> 2.2"
```hcl
terraform {
  required_version = ">= 0.13"
  required_providers {
    google = {
      source  = "hashicorp/google"
      version = ">= 3.50, < 4.0"
    }
    google-beta = {
      source  = "hashicorp/google-beta"
      version = ">= 3.50, < 4.0"
    }
    null = {
      source  = "hashicorp/null"
      version = "~> 2.1"
    }
    random = {
      source  = "hashicorp/random"
      version = "~> 2.2"
```

### Terraform Version

```sh
Same error is seen with TF version 0.12.16 and 0.13.15
```

**Comment by bharathkkb** [comment]
Hi @anitagutta 
Closing this since we debugged offline as an issue with Terraform version mismatch in TFE
